frontend/src/lib/terminal.js

```diff
@@ -152,6 +154,27 @@ export class TerminalSession {
       if (this.cancelConnect) {
         return
       }
+
+      if (!this.workaroundOnce) {
```

Comment:
I would propose to move the workaround code into function `waitForPodRunning` by wrapping the the connectPromise code into a function that return an open websocket connection.
```js
async function waitForPodRunning (hostCluster, containerName, handleEvent, timeoutSeconds) {
  const connectTimeoutSeconds = 5

  const createConnection = () => {
    const protocols = ['garden'] // there must be at least one other subprotocol in addition to the bearer token
    addBearerToken(protocols, hostCluster.token)
    const url = watchPodUri(hostCluster)
    const ws = new WebSocket(url, protocols)

    const connectPromise = new Promise((resolve, reject) => {
      const openHandler = () => {
        ws.removeEventListener('open', openHandler)
        ws.removeEventListener('error', errorHandler)

        resolve(ws)
      }
      const errorHandler = ev => {
        ws.removeEventListener('open', openHandler)
        ws.removeEventListener('error', errorHandler)

        reject(new Error(`Failed to create connection to ${url}`))
      }

      ws.addEventListener('open', openHandler)
      ws.addEventListener('error', errorHandler)
    })

    return pTimeout(connectPromise, connectTimeoutSeconds * 1000, `Could not connect within ${connectTimeoutSeconds} seconds`)
  }

  let ws = null
  try {
    ws = await createConnection()
  } catch (err) {
    logger.info(err.message)
    ws = null
  }

  if (!ws) {
    // workaround for https://bugs.chromium.org/p/chromium/issues/detail?id=993907&q=ERR_SSL_CLIENT_AUTH_CERT_NEEDED&can=2
    logger.info('Trying connect workaround ...')
    try {
      // We want the tls handshake to be establised so that the request context has a client certificate selection for the endpoint.
      // It does not matter that the request will fail with 401.
      const options = {
        method: 'HEAD',
        mode: 'no-cors',
      }
      await fetch(`https://${hostCluster.kubeApiServer}`, options)
    } catch (err) { /* ignore errors */ }
    ws = await createConnection()
  }

  const podRunningPromise = new Promise((resolve, reject) => {
    ... 
  })

  try {
    await pTimeout(podRunningPromise, timeoutSeconds * 1000, `Timed out after ${timeoutSeconds}s`)
  } finally {
    closeWsIfNotClosed(ws)
  }
}
```

This first parameter of this function is now the hostCluster and not the websocket. This must be adapted. I would also move the finally handling into this function. We could also switch to long polling as a workaround. 
```js

  async waitUntilPodIsRunning (timeoutSeconds) {
    ...
    return waitForPodRunning(this.hostCluster, containerName, onPodStateChange, timeoutSeconds * 1000)
  }
```

Reply:
✅